### PR TITLE
[APPC-3704] Paypal v2 send user to new login, error handling

### PR DESCRIPTION
**Pull request by rodolfobcardoso**
**What does this PR do?**
When the billing agreement is not valid, then sends the user to do another login again. 

**Database changed?**
No

**Where should the reviewer start?**
Check the last commit.

**How should this be manually tested?**
Use a paypal account without funds 

**What are the relevant tickets?**
https://aptoide.atlassian.net/browse/APPC-3704

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] Database migration (if applicable)?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/billing/paypal/PayPalIABViewModel.kt [hunk 1]
```diff
@@ -119,13 +128,16 @@ class PayPalIABViewModel @Inject constructor(
         .subscribeOn(networkScheduler)
         .observeOn(viewScheduler)
         .doOnSuccess {
-          Log.d(TAG, "Successful Token creation ")  //TODO event
+          Log.d(TAG, "Successful Token creation ")
```
Comment (giovanni-aptoide) [inline app/src/main/java/com/asfoundation/wallet/billing/paypal/PayPalIABViewModel.kt:131]: why keep this log?

File: app/src/main/java/com/asfoundation/wallet/billing/paypal/PayPalIABViewModel.kt [hunk 2]
```diff
@@ -119,13 +128,16 @@ class PayPalIABViewModel @Inject constructor(
...
           authenticatedToken = it.token
           _state.postValue(State.WebViewAuthentication(it.redirect.url))
         }
         .subscribe({}, {
           Log.d(TAG, it.toString())
-          sendPaymentErrorEvent("Error on token creation", transactionBuilder)
+          sendPaymentErrorEvent(
+            errorMessage = "Error on token creation",
+            transactionBuilder = transactionBuilder
```
Comment (giovanni-aptoide) [inline app/src/main/java/com/asfoundation/wallet/billing/paypal/PayPalIABViewModel.kt:139]: Couldn’t you move these strings for strings.xml

File: app/src/main/java/com/asfoundation/wallet/billing/paypal/repository/PayPalV2Repository.kt [hunk 3]
```diff
@@ -151,22 +155,25 @@ class PayPalV2Repository @Inject constructor(
...
     val validity = when (errorCode) {
       404 -> PaypalTransaction.PaypalValidityState.NO_BILLING_AGREEMENT
+      400 -> PaypalTransaction.PaypalValidityState.NO_BILLING_AGREEMENT
       else -> PaypalTransaction.PaypalValidityState.NO_BILLING_AGREEMENT
-      // Until all payment errors are treated, if the payment fails with a previous billing
-      // agreement, then alway tries to login again once
+      // If the payment fails with a previous billing agreement, then always tries to login
+      // again once
 //      else -> PaypalTransaction.PaypalValidityState.ERROR
     }
```
Comment (giovanni-aptoide) [inline app/src/main/java/com/asfoundation/wallet/billing/paypal/repository/PayPalV2Repository.kt:169]: why keep these commented code?

File: core/network/microservices/src/main/java/com/appcoins/wallet/core/network/microservices/api/broker/PaypalV2Api.kt [hunk 4]
```diff
@@ -4,13 +4,15 @@ import com.appcoins.wallet.core.network.microservices.model.*
...
+import retrofit2.http.Header
 import retrofit2.http.POST
 import retrofit2.http.Query
 
 interface PaypalV2Api {
 
   @POST("8.20230522/gateways/paypal/transactions")
   fun createTransaction(
+    @Header("PayPal-Mock-Response") mockHeader: String, // uncomment for testing errors in dev
     @Query("wallet.address") walletAddress: String,
```
Comment (giovanni-aptoide) [inline core/network/microservices/src/main/java/com/appcoins/wallet/core/network/microservices/api/broker/PaypalV2Api.kt:16]: Aren’t there any way to have it managed by BuildConfig?

**Review by giovanni-aptoide — APPROVED** [review]

**Review by gustavoaptoide — APPROVED** [review]

**Pull request merged**